power_controller.php

```diff
@@ -48,15 +48,15 @@ public function get_data($serial_number = '')
      **/
     public function get_stats()
     {
-        $out = array();
+        $obj = new View();
+        
         if (! $this->authorized()) {
```

Comment:
You can get rid of all the $this-authorized() calls. All methods are now protected except get_script() and index()